test/unit/reductions/exponentiated_gradient/test_lagrangian.py

```diff
@@ -119,3 +120,40 @@ def test_call_oracle(Constraints, eps, mocker):
     assert (kwargs['sample_weight'] == redW).all()
     assert lagrangian.n_oracle_calls == 1
     assert len(lagrangian.oracle_execution_times) == 1
+
+
+@pytest.mark.parametrize("Constraints", [DemographicParity, EqualizedOdds])
```

Comment:
This is okay, but I'm a bit worried about the growing copy-paste overlap among all the tests of the Lagrangian. I wonder if there's a good way to refactor. But maybe better to leave that for a future PR.

Reply:
I would agree that test refactoring is something which needs doing. Although the thing which triggered that realisation was needing to identify which of the twenty-or-so dictionaries in the smoke-test data required changing. I do think it's better to leave that for a future PR, when we're _only_ changing the tests, and not the underlying functionality.

Reply:
absolutely!